### PR TITLE
[Candidate List] Update User::singleton() declarations to facilitate unit testing

### LINKED ISSUE #5015: Update User::singleton() declarations
Code that declares a User object like this:

`$user = User::singleton();`

should be changed to:
```
$factory = \NDB_Factory::singleton();
$user = $factory->user();
```
or

`$user  = \NDB_Factory::singleton()->user();`

Using the factory class allows for easier and more robust testing.

### PR DESCRIPTION
**Pull request by laemtl**
## Brief summary of changes

Replaced occurence of 
$user = \User::singleton();
$db = \Database::singleton();

by

$factory  = \NDB_Factory::singleton();
$db       = $factory->database();
$user     = $factory->user(); 

according to the unit test guide (UnitTestGuide.md).

#### Link(s) to related issue(s)

* Resolves #5015

#### About me
GOSC applicant, happy to join the community!
This is my first contribution, don't hesitate to comment on the possible improvements.

@christinerogers

### CODE REVIEW

File: modules/candidate_list/php/candidate_list.class.inc [hunk 1]
```diff
@@ -74,8 +74,9 @@ class Candidate_List extends \DataFrameworkMenu
         \Module::factory('candidate_parameters');
 
         // create user object
-        $user   = \User::singleton();
-        $config = \NDB_Config::singleton();
+        $factory = \NDB_Factory::singleton();
+        $user    = $factory->user();
+        $config  = \NDB_Config::singleton();
```
Comment (maltheism) [inline modules/candidate_list/php/candidate_list.class.inc]: Hi @laemtl, you can also do `$factory->config();` for accessing the config. @johnsaigle showed me that.

Comment (laemtl) [reply modules/candidate_list/php/candidate_list.class.inc]: Hello @maltheism,

Thanks for pointing this out! 
I noticed the config method in /NDB_Factory.class.inc and was curious why it was not referenced anywhere in the unit testing guide. It's definitely a good idea to uniformize everything at once. I will review all the submitted modules.

Comment (johnsaigle) [reply modules/candidate_list/php/candidate_list.class.inc]: Thanks for the PR @laemtl, this is a great improvement. If you're up for it amending the Unit Test Guide would be really helpful also. If not I'll make a separate issue for it.

Comment (johnsaigle) [inline modules/candidate_list/php/candidate_list.class.inc]:
```suggestion
        $config  = $factory->config();
```

I'll approve the PR after this change.

**Comment by johnsaigle** [comment]
Really close, just a couple of formatting problems reported by our automated testing:

> FILE: ...ces/Loris/modules/candidate_list/php/candidatelistrowprovisioner.class.inc  
39 | ERROR | [x] Equals sign not aligned with surrounding assignments;
    |       |     expected 2 spaces but found 1 space
 40 | ERROR | [x] Equals sign not aligned with surrounding assignments;
    |       |     expected 3 spaces but found 2 spaces

This can be replicated locally if you run `make checkstatic`. Running the following command should automatically fix the formatting issues:

```bash
vendor/bin/phpcbf --standard=test/LorisCS.xml modules/candidate_list/php/candidatelistrowprovisioner.class.inc
```

**Comment by laemtl** [comment]
Yes, sorry for that, I should remember to run ./test/run-php-linter.sh locally before making any update.

**Comment by laemtl** [comment]
@johnsaigle I also updated the Unit Test Guide in the Core PR.

**Pull request merged**